### PR TITLE
fix node gc

### PR DESCRIPTION
**Pull request by zhangzujian**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

- Bug fixes

## Which issue(s) this PR fixes

When a node where kube-ovn-controller is running is deleted, or a node is deleted during kube-ovn-controller is down, the logical router policy with priority 30000 is not deleted after kube-ovn-controller is started.

### CODE REVIEW

File: pkg/controller/node.go [hunk 1]
```diff
@@ -987,13 +976,25 @@ func (c *Controller) checkPolicyRouteExistForNode(nodeName, cidr, nexthop string
 	return false, nil
 }
 
-func (c *Controller) deletePolicyRouteForNode(nodeName string) error {
+func (c *Controller) deletePolicyRouteForNode(nodeName, portName string) error {
 	subnets, err := c.subnetsLister.List(labels.Everything())
```
Comment (bobz965) [inline pkg/controller/node.go:980]: 提个问题，这里是不是只有默认vpc下的subnet？

File: pkg/controller/gc.go [hunk 2]
```diff
@@ -219,30 +219,44 @@ func (c *Controller) gcNode() error {
...
 		klog.Errorf("failed to list ip, %v", err)
 		return err
 	}
-	ipNodeNames := make([]string, 0, len(ips))
+
 	for _, ip := range ips {
-		if !strings.Contains(ip.Name, ".") {
-			ipNodeNames = append(ipNodeNames, strings.TrimPrefix(ip.Name, "node-"))
+		if strings.HasPrefix(ip.Name, util.NodeLspPrefix) && !strings.Contains(ip.Name, ".") {
+			if node := ip.Name[len(util.NodeLspPrefix):]; !nodeNames.Has(node) {
```
[the remark this answers is not in the document]
Comment (zhangzujian) [reply pkg/controller/gc.go:234]: > The expression len(s) is [constant](https://go.dev/ref/spec#Constants) if s is a string constant.

https://go.dev/ref/spec#Length_and_capacity

File: pkg/controller/node.go (hunk 1 above)
Comment (zhangzujian) [reply pkg/controller/node.go:980]: 只删除默认 VPC 下的策略路由。

File: pkg/controller/gc.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bobz965) [reply pkg/controller/gc.go:234]: ok

**Review by bobz965 — APPROVED** [review]

**Pull request merged**